src/frontend/js/components/RouterButton/index.tsx

```diff
@@ -0,0 +1,30 @@
+import { MouseEvent, useRef } from 'react';
+import { useHref, useNavigate } from 'react-router-dom';
+import { Button, ButtonProps } from 'components/Button';
+import { location } from 'utils/indirection/window';
+
+export const RouterButton = (props: ButtonProps) => {
+  if (!props.href) {
+    throw new Error("The attribute 'href' is required when using RouterButton.");
+  }
+  const hrefBase = useHref('');
+  const navigate = useNavigate();
+  const ref = useRef<HTMLButtonElement & HTMLAnchorElement>(null);
+  return (
+    <Button
+      ref={ref}
+      href={hrefBase + props.href}
+      onClick={(event: MouseEvent) => {
+        event.preventDefault();
+        if (props.href?.match('^http')) {
+          // We must do this to be able to tests redirects.
+          location.replace(props.href);
+          return;
+        }
+        navigate(props.href!);
+        ref.current?.blur();
+      }}
```

Comment:
In fact, if `props.href?.match('^http')` we could simply execute nothing.

```suggestion
onClick={(event: MouseEvent) => {
        const isDashboardRoute = !props.href?.match('^http');
        if (!isDashboardRoute) return;

        event.preventDefault();
        navigate(props.href!);
        ref.current?.blur();
      }}
```

Reply:
`navigate` has not been called ?

Reply:
I did the change, but unfortunately this forced me to mock it also for the first test ( "navigates inside the router" ) because I didn't succeed to mock it ONLY for the second test. 

So the consequence is that we lost the real navigation test during the first test. 

I spent the morning on it and I couldn't find a way to do it, so if you're willing to do a pair programming to find a way to do it, I'm you're guy because I tried everything Jest provides .. 😅